aperturedb/ParallelLoader.py

```diff
@@ -16,6 +16,59 @@ def __init__(self, db, dry_run=False):
         super().__init__(db, dry_run=dry_run)
         self.type = "element"
 
+    def get_existing_indices(self):
+        schema_result, _ = self.db.query([{"GetSchema": {}}])
+        schema = schema_result[0]["GetSchema"]
+        existing_indices = {}
+        if schema:
+            for index_type in (("entity", "entities"), ("connection", "connections")):
+                foo = schema.get(index_type[1]) or {}
+                bar = foo.get("classes") or {}
+                for cls_name, cls_schema in bar.items():
+                    props = cls_schema.get("properties") or {}
+                    for prop_name, prop_schema in props.items():
+                        if prop_schema[1]:  # indicates property has an index
+                            existing_indices.setdefault(index_type[0], {}).setdefault(
+                                cls_name, set()).add(prop_name)
+        return existing_indices
+
+    def create_indices(self, indices):
+        if len(indices) == 0:
+            return
+
+        existing_indices = self.get_existing_indices()
+        new_indices = []
+        for tp, classes in indices.items():
+            ex_classes = existing_indices.get(tp, {})
+            for cls, props in classes.items():
+                ex_props = ex_classes.get(cls, set())
+                for prop in props - ex_props:
+                    new_indices.append({
+                        "index_type": tp,
+                        "class": cls,
+                        "property_key": prop
+                    })
+
+        if len(new_indices) == 0:
+            return
+
+        logger.info(
+            f"Creating {len(new_indices)} indices: {new_indices}.")
+
+        create_indices = [{"CreateIndex": idx} for idx in new_indices]
+
+        res, _ = self.db.query(
```

Comment:
same here, we should just use [this method on Utils.py](https://github.com/aperture-data/aperturedb-python/blob/develop/aperturedb/Utils.py#L207).